[bryansendeavour] iP

bryansendeavour opened the pull request:
#Duke (in development stages)
> "ahissi jinja" -oppa

Duke helps u keep track of your:
- Events
- Todo
- ~~due~~Deadlines 

All you need to do is:
1. Look at source code [here](https://github.com/bryansendeavour/ip)
2. See how it works
3. Try it out by running it :rofl:

and it is **Free lor**. Waiting for semo? 

Features:
- [X] See your tasks
- [X] Save your tasks

---
Run this to make it work
```
public static void main(String[] args) throws Exception {
     new Duke(DIRPATH, FILEPATH).run();
}
```

Commit:
Add Gradle support

bryansendeavour committed:
Level-1

bryansendeavour committed:
Level-2

bryansendeavour committed:
Level-3

bryansendeavour committed:
Level-4

bryansendeavour committed:
Some exception handling

bryansendeavour committed:
Added delete

bryansendeavour committed:
no message

bryansendeavour committed:
Revert "no message"

This reverts commit ffa7764ffb2599fc3a4522d7bda87e707c9cffc3.

bryansendeavour committed:
AUI-Testing

bryansendeavour committed:
Week 2

bryansendeavour committed:
Merge commit 'c8c1e0964f2713f7e7d75730e3ca9eac6f91a418'

bryansendeavour committed:
Update for Level-5

bryansendeavour committed:
Change of implementation and save feature addition

bryansendeavour committed:
Add branch-Level-8 implementations

bryansendeavour committed:
Add A-MoreOOP

bryansendeavour committed:
Add A-Packages

bryansendeavour committed:
Merge branch 'add-gradle-support' of ssh://github.com/bryansendeavour/ip into add-gradle-support

bryansendeavour committed:
Add A-Gradle

bryansendeavour committed:
no message

bryansendeavour committed:
no message

bryansendeavour committed:
Addition of JavaDoc

bryansendeavour committed:
Add A-CodingStandard

bryansendeavour committed:
Add Level-9

noahxinjie commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,99 @@
+
+    private final String NAME;
+    static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yy HHmm");
+    private boolean done;
```
Perhaps naming the boolean as isDone instead might be better?

noahxinjie commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,160 @@
+    }
+
+    /**
+     * This method removes the specified task.
+     * @param num serial number of task.
```
I think it might be better to put an empty line between the description and parameter section of your JavaDocs  ^^

noahxinjie commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,53 @@
+public class Storage {
+
+    private static String DIRPATH;
+    private static String FILEPATH;
```
Perhaps it might be better to make these two attributes final too

noahxinjie reviewed (commented):
I like your code! It is generally clean and easy to understand and follows the coding standards :)

JThh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,53 @@
+     * for future reference.
+     * @throws Exception when file already exist.
+     */
+    private void init() throws Exception {
```
Can the name of this method be more descriptive? That will be more helpful to understand its intended use!

JThh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,163 @@
+        return sc.nextLine();
+    }
+
+    /**
```
Would it be more succinct to shrink below print functions into one and differentiate the print messages using if-else?

JThh reviewed (changes requested):
Overall, the codes are well-written and carefully modularized.

Shuggan commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    private String[] inputs;
+
+    private static StringBuilder strBuild = new StringBuilder();
```
Instead of using a static string builder, consider making it a non-static instance variable and creating a new instance of it in the find method.

Shuggan commented on src/main/java/task/TaskList.java:
```diff
@@ -98,9 +98,44 @@ public static void printList() {
+    /**
+     * This method writes the tasks to data file.
+     */
     public static void writeToFile() {
```
In the writeToFile() method, the code for writing an empty file is duplicated in the case of an empty task list. This can possibly be refactored to a separate method to improve readability

bryansendeavour committed:
Add GUI implementations

bryansendeavour committed:
Add Assertions

The methods for these commands are assumed to work accordingly

Might encounter some bugs where command is executed when not specified to do so

Let's add Assertions to double check the correct command was used

Assertions is one way to check the accuracy

bryansendeavour committed:
Change mainClassName to MainLauncher

bryansendeavour committed:
Tweak some features and handle more exceptions

Code needs to be more concise and readable together with bug fixes

Let's brush up the code and create more methods whenever possible with good naming conventions.
Let's add more exception handling.

Using better naming conventions is preferred for readability

bryansendeavour committed:
Merge pull request #1 from bryansendeavour/branch-A-Assertions

Branch a assertions

bryansendeavour committed:
Merge branch 'master' into branch-A-CodeQuality

bryansendeavour committed:
Merge pull request #2 from bryansendeavour/branch-A-CodeQuality

Branch a code quality

JavonTeo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,47 @@
+                return new DeleteCommand(splitCommand[1]).delete();
+            case "FIND" :
+                return new FindCommand(splitCommand).find();
+            case "BYE" :
```
It might be possible to abstract the case "BYE" into a separate command class. This would allow this segment of the code to have SLAP!

JavonTeo commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,62 @@
+        if (inputs.length == 1) {
+            throw new DukeException(Ui.printMissingName());
+        } else {
+            boolean isName = true;
```
It might be possible to prevent nested code by having the else block be part of the main code, instead of an else clause.

JavonTeo reviewed (commented):
Overall good quality code, just some nits to fix.

bryansendeavour committed:
Add Undo command

bryansendeavour committed:
User guide

bryansendeavour committed:
no message

bryansendeavour committed:
no message
